src/test/java/org/retest/rebazer/connector/GithubConnectorTest.java

```diff
@@ -139,10 +144,221 @@ void getAllPullRequests_should_return_all_pull_requests_as_list() throws Excepti
 	@Test
 	void getLatestUpdate_should_return_updated_PullRequest() {
 		final PullRequest pullRequest = mock( PullRequest.class );
-		final String json = "{\"updated_at\": \"someTimestamp\"}";
-		when( template.getForObject( anyString(), eq( String.class ) ) ).thenReturn( json );
+		final String repositoryTime = "{\"updated_at\": \"2019-01-04T15:00:50Z\"}";
+		when( template.getForObject( anyString(), eq( String.class ) ) ).thenReturn( repositoryTime );
+		final Date lastUpdate = Date.from( OffsetDateTime.parse( "2010-01-01T00:00:00Z" ).toInstant() );
+		when( pullRequest.getLastUpdate() ).thenReturn( lastUpdate );
+
+		final String checksFinished =
+				"{\"check_runs\":[{\"completed_at\":\"2019-01-04T15:30:50Z\"},{\"completed_at\":\"2019-01-04T15:30:59Z\"},{\"completed_at\":\"2019-01-04T15:29:59Z\"}]}";
+		final ResponseEntity<String> resp = new ResponseEntity<>( checksFinished, HttpStatus.OK );
+		when( template.exchange( anyString(), any( HttpMethod.class ), any( HttpEntity.class ), eq( String.class ) ) )
+				.thenReturn( resp );
+
+		assertThat( cut.newestChecksTime( pullRequest ) ).isEqualTo( "2019-01-04T15:30:59Z" );
+		assertThat( cut.getLatestUpdate( pullRequest ).getLastUpdate() ).hasSameTimeAs( "2019-01-04T15:30:59Z" );
+
+	}
+
+	@Test
+	void newestChecksTime_should_return_newest_time_of_the_checks() {
+		final PullRequest pullRequest = mock( PullRequest.class );
+		final String checksFinished1 =
```

Comment:
To keep tests simple, one shouldn't test multiple aspects within a single test case. `checksFinished1`, `checksFinished2`, and `checksFinished3` look unrelated. If it doesn't make sense to extract separate test case as it is just different test data, have a look at [parameterized tests](https://baeldung.com/parameterized-tests-junit-5/). (Same goes for the other tests below.)